src/info/manifest.c

```diff
@@ -0,0 +1,1986 @@
+            else if (strEq(key, MANIFEST_KEY_OPTION_ARCHIVE_COPY_STR))
+                manifest->data.backupOptionArchiveCopy = jsonToBool(value);
+            else if (strEq(key, MANIFEST_KEY_OPTION_COMPRESS_STR))
+                manifest->data.backupOptionCompress = jsonToBool(value);
+            else if (strEq(key, MANIFEST_KEY_OPTION_HARDLINK_STR))
+                manifest->data.backupOptionHardLink = jsonToBool(value);
+            else if (strEq(key, MANIFEST_KEY_OPTION_ONLINE_STR))
+                manifest->data.backupOptionOnline = jsonToBool(value);
+
+            // Options that were added after v1.00 and may not be present in every manifest
+            else if (strEq(key, MANIFEST_KEY_OPTION_BACKUP_STANDBY_STR))
+                manifest->data.backupOptionStandby = varNewBool(jsonToBool(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_BUFFER_SIZE_STR))
+                manifest->data.backupOptionBufferSize = varNewUInt(jsonToUInt(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_CHECKSUM_PAGE_STR))
+                manifest->data.backupOptionChecksumPage = varNewBool(jsonToBool(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_COMPRESS_LEVEL_STR))
+                manifest->data.backupOptionCompressLevel = varNewUInt(jsonToUInt(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_COMPRESS_LEVEL_NETWORK_STR))
+                manifest->data.backupOptionCompressLevelNetwork = varNewUInt(jsonToUInt(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_DELTA_STR))
+                manifest->data.backupOptionDelta = varNewBool(jsonToBool(value));
+            else if (strEq(key, MANIFEST_KEY_OPTION_PROCESS_MAX_STR))
+                manifest->data.backupOptionProcessMax = varNewUInt(jsonToUInt(value));
+        }
+        MEM_CONTEXT_END();
+    }
+
+    FUNCTION_TEST_RETURN_VOID();
+}
+
+Manifest *
+manifestNewLoad(IoRead *read)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(IO_READ, read);
+    FUNCTION_LOG_END();
+
+    ASSERT(read != NULL);
+
+    Manifest *this = NULL;
+
+    MEM_CONTEXT_NEW_BEGIN("Manifest")
+    {
+        this = manifestNewInternal();
+
+        // Load the manifest
+        ManifestLoadData loadData =
+        {
+            .memContext = memContextNew("load"),
+            .manifest = this,
+        };
+
+        MEM_CONTEXT_BEGIN(loadData.memContext)
+        {
+            loadData.fileFoundList = lstNew(sizeof(ManifestLoadFound));
+            loadData.linkFoundList = lstNew(sizeof(ManifestLoadFound));
+            loadData.pathFoundList = lstNew(sizeof(ManifestLoadFound));
+        }
+        MEM_CONTEXT_END();
+
+        this->info = infoNewLoad(read, manifestLoadCallback, &loadData);
+
+        // Process file defaults
+        for (unsigned int fileIdx = 0; fileIdx < lstSize(this->fileList); fileIdx++)
+        {
+            ManifestFile *file = lstGet(this->fileList, fileIdx);
```

Comment:
For consistency, would it be better to use manifestFileTotal(this) and manifestFile(this, fileIdx) instead of lstSize and lstGet? If so, this function should also change for link and path. Not a big deal if don't want to change.

Reply:
Using *Total() in place of lstSize() everywhere now.

Using manifestFile(), etc. when possible but since that returns a const pointer we still need to use lstGet() in the cases where we need to modify the struct.  The case above is an example of this.